src/Combinatorics/Graphs/functions.jl

```diff
@@ -5,6 +5,24 @@ function pm_object(G::Graph{T}) where {T <: Union{Directed, Undirected}}
   return G.pm_graph
 end
 
+function Base.:(==)(a::Graph{T}, b::Graph{T}) where {T <: Union{Directed, Undirected}}
+  return adjacency_matrix(a) == adjacency_matrix(b)
+end
+
+function Base.:(==)(a::MixedGraph, b::MixedGraph)
+  _directed_component(a) == _directed_component(b) &&
+    _undirected_component(a) == _undirected_component(b)
+end
+
+function Base.hash(g::Graph, h::UInt)
+  return hash(adjacency_matrix(g), h)
+end
+
+function Base.hash(g::MixedGraph, h::UInt)
+  return hash(adjacency_matrix(_directed_component(g)),
+              hash(adjacency_matrix(_undirected_component(g)), h))
+end
```

Comment:
is there a particular reason you mention the adjacency matrices here and not just do
```suggestion
function Base.hash(g::MixedGraph, h::UInt)
  return hash(_directed_component(g), hash(_undirected_component(g), h))
end
```

Reply:
no